src/Misc/Sync.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
```

Comment:
Shouldn't we use the usual header guards instead of `#pragma`, which might not be supported by some compilers?

Also, I miss a copyright header.

Reply:
#pragma was copy&paste from Time.h ;-)